packages/cli/src/commands/setup/vite/vite.js

```diff
@@ -0,0 +1,154 @@
+
+import { addWebPackages } from '@redwoodjs/cli-helpers'
+import { getConfigPath } from '@redwoodjs/internal/dist/paths'
+import { errorTelemetry } from '@redwoodjs/telemetry'
+
+import { getPaths, transformTSToJS, writeFile } from '../../../lib'
+import c from '../../../lib/colors'
+import { isTypeScriptProject } from '../../../lib/project'
+
+export const command = 'vite'
+export const description =
+  '[Experimental] Configure the web side to use Vite, instead of Webpack'
+export const builder = (yargs) => {
+  yargs.option('force', {
+    alias: 'f',
+    default: false,
+    description: 'Overwrite existing configuration',
+    type: 'boolean',
+  })
+  yargs.option('verbose', {
+    alias: 'v',
+    default: false,
+    description: 'Print more logs',
+    type: 'boolean',
+  })
+  yargs.option('add-package', {
+    default: true,
+    description:
+      'Allows you to skip adding the @redwoodjs/vite package. Useful for testing',
+    type: 'boolean',
+  })
+}
+
+export const handler = async ({ force, verbose, addPackage }) => {
+  const ts = isTypeScriptProject()
+  const tasks = new Listr(
+    [
+      {
+        title: 'Confirmation',
+        task: async (_ctx, task) => {
+          const confirmation = await task.prompt({
+            type: 'Confirm',
+            message: 'Vite support is experimental. Continue?',
+          })
+
+          if (!confirmation) {
+            throw new Error('User aborted')
+          }
+        },
+      },
+      {
+        title: 'Adding vite.config.js...',
+        task: () => {
+          const viteConfigPath = `${getPaths().web.base}/vite.config.${
+            ts ? 'ts' : 'js'
+          }`
+
+          const templateContent = fs.readFileSync(
+            path.resolve(__dirname, 'templates', 'vite.config.ts.template'),
+            'utf-8'
+          )
+
+          const viteConfigContent = ts
+            ? templateContent
+            : transformTSToJS(viteConfigPath, templateContent)
+
+          return writeFile(viteConfigPath, viteConfigContent, {
+            overwriteExisting: force,
+          })
+        },
+      },
+      {
+        title: 'Adding Vite bundler flag to redwood.toml...',
+        task: (_ctx, task) => {
+          const redwoodTomlPath = getConfigPath()
+          const config = toml.parse(fs.readFileSync(redwoodTomlPath, 'utf-8'))
+
+          if (config.web.bundler !== 'vite') {
+            config.web.bundler = 'vite'
+
+            writeFile(redwoodTomlPath, toml.stringify(config), {
+              overwriteExisting: true, // redwood.toml always exists
+            })
+          } else {
+            task.skip('Vite bundler flag already set in redwood.toml')
+          }
+        },
+      },
+      {
+        title: 'Creating new entry point in `web/src/entry-client.jsx`...',
+        task: () => {
+          // Keep it as JSX for now
+          const entryPointFile = path.join(
+            getPaths().web.src,
+            `entry-client.jsx`
```

Comment:
Not sure about the name here; why not just call it `index.js` like the one we already have? And this file has to be there? Right now we keep `index.js` under wraps, and give users the option to generate it if they want more power. What's the nuance or reasoning for changing?

Reply:
I've intentionally called it entry-client, because for render modes you need a different entry point for server rendering. 

I know I originally hid the index.js, but in retrospect I think it doesn't change as often as I thought it did. We also didn't have codemod capability at the time!